Add a list of plugins and note how they are used (#7)
canonical/snap-tempest: README.md

The tempest snap can be installed directly from the snap store:

    sudo snap install [install-OPTIONS] tempest

## Plugins

The snap provides all the plugins listed in the [OpenStack release notes] and those
cannot be removed or updated. Currently, it is not possible to add new plugins or use
your own.

List of plugins:

- barbican-tempest-plugin
- blazar-tempest-plugin
- cinder-tempest-plugin
- cloudkitty-tempest-plugin
- cyborg-tempest-plugin
- designate-tempest-plugin
- ec2api-tempest-plugin
- freezer-tempest-plugin
- glance-tempest-plugin
- heat-tempest-plugin
- ironic-tempest-plugin
- keystone-tempest-plugin
- kuryr-tempest-plugin
- magnum-tempest-plugin
- manila-tempest-plugin
- mistral-tempest-tests
- monasca-tempest-plugin
- murano-tempest-plugin
- neutron-tempest-plugin
- octavia-tempest-plugin
- oswin-tempest-plugin
- sahara-tests
- senlin-tempest-plugin
- solum-tempest-plugin
- telemetry-tempest-plugin
- trove-tempest-plugin
- venus-tempest-plugin
- vitrage-tempest-plugin
- watcher-tempest-plugin
- zaqar-tempest-plugin
- zun-tempest-plugin

## Quickstart
To run Tempest, please follow the steps in the [Tempest QuickStart] documentation.

> The tempest snap provides a set of pre-defined test lists with different focuses.

Reviewer: well, only one for now...

Reviewer 2: As discussed elsewhere, we are keeping this wording plural because more test lists are going to be added in the following PRs.

For quickly verifying the behavior of the target cloud and making sure tempest is
correctly set up, you can run tests listed in the `readonly-quick.txt` file:

    tempest run --load-list '@BUILTIN_TESTLISTS/readonly-quick.txt' 

The `@BUILTIN_TESTLISTS` keyword will be automatically replaced by the full path
of the directory storing the built-in test lists.

**Note**: the single-quotation marks around the test list file path is required

> if the `@BUILTIN_TESTLISTS` keyword is used (in other words, when running tempest

Reviewer: ```suggestion
If the `@BUILTIN_TESTLISTS` keyword is used (in other words, when running tempest
```

Reviewer 2: As discussed elsewhere, `if` should be kept lower-cased.

The tempest snap can be installed directly from the snap store:

    sudo snap install --edge tempest

## Plugins

> The image is baked with all plugins listed in the release notes and those cannot be

Reviewer 2: nit: it would be better to provide a link to the release notes you mentioned.

>     sudo snap install --edge tempest

Reviewer: This README goes into the master branch so I'm thinking it'd be better to be more generic and don't assume `--edge`

## Plugins

The image is baked with all plugins listed in the [OpenStack release notes] and those
cannot be removed or updated. Currently, it is not possible to add new plugins or use
your own.

List of plugins:

- barbican-tempest-plugin
- blazar-tempest-plugin
- cinder-tempest-plugin
- cloudkitty-tempest-plugin
- cyborg-tempest-plugin
- designate-tempest-plugin
- ec2api-tempest-plugin
- freezer-tempest-plugin
- glance-tempest-plugin
- heat-tempest-plugin
- ironic-tempest-plugin
- keystone-tempest-plugin
- kuryr-tempest-plugin
- magnum-tempest-plugin
- manila-tempest-plugin
- mistral-tempest-tests
- monasca-tempest-plugin
- murano-tempest-plugin
- neutron-tempest-plugin
- octavia-tempest-plugin
- oswin-tempest-plugin
- sahara-tests
- senlin-tempest-plugin
- solum-tempest-plugin
- telemetry-tempest-plugin
- trove-tempest-plugin
- venus-tempest-plugin
- vitrage-tempest-plugin
- watcher-tempest-plugin
- zaqar-tempest-plugin
- zun-tempest-plugin

## Quickstart

> To run Tempest, please follow the steps in [Tempest QuickStart] documentation.

Reviewer: ```suggestion
To run Tempest, please follow the steps in the [Tempest QuickStart] documentation.
```

The tempest snap provides a set of pre-defined test lists with different focuses.

> For quickly verifying the behavior of the objective cloud and making sure tempest

Reviewer: ```suggestion
For quickly verifying the behavior of the target cloud and making sure tempest
```

is correctly set up, you can run tests listed in the `readonly-quick.txt` file:

    tempest run --load-list '@BUILTIN_TESTLISTS/readonly-quick.txt' 

> `@BUILTIN_TESTLISTS` keyword will be automatically replaced by the full path of

Reviewer: ```suggestion
The `@BUILTIN_TESTLISTS` keyword will be automatically replaced by the full path of
```

the directory storing the built-in test lists.

**Note**: the single-quotation marks around the test list file path is required

> if `@BUILTIN_TESTLISTS` keyword is used (in another words, when running tempest

Reviewer: ```suggestion
If the `@BUILTIN_TESTLISTS` keyword is used (in other words, when running tempest
```

> with a built-in test list file). Fail to do that will result in wrong path error.

Reviewer: ```suggestion
with a built-in test list file), failure to do so will result in a wrong path error.
```

## This snap includes

### python-tempestconf

> This snapshot includes [python-tempestconf], which automatically generates a tempest

Reviewer: ```suggestion
This snap includes [python-tempestconf], which automatically generates a tempest
```

> configuration based on your cloud.

Reviewer: ```suggestion
configuration appropriate for your cloud.
```

> It provided single CLI options [`discover-tempest-config`] which is provided in snap

Reviewer 2: ```suggestion
It provided a single CLI option [`discover-tempest-config`] which is provided in snap
```

> It provided single CLI options [`discover-tempest-config`] which is provided in snap

Reviewer: ```suggestion
It provides a single CLI command [`discover-tempest-config`] which is aliased in the snap
```

> via `tempest.discover`.

Reviewer: ```suggestion
as `tempest.discover`.
```

> [OpenStack release notes]: https://releases.openstack.org/antelope/index.html#tempest-plugins

Reviewer: too bad there's no `https://releases.openstack.org/latest/index.html#tempest-plugins`...

Reviewer 2: Yes, unfortunately we have to update this each time a new os release is out.
